lib/util/url.spec.ts

```diff
@@ -6,57 +6,91 @@ import {
+    ${'http://foo.io/aaa/'} | ${'bbb'}                | ${'http://foo.io/aaa/bbb'}
+    ${'http://foo.io/aaa'}  | ${'/bbb/'}              | ${'http://foo.io/aaa/bbb/'}
+    ${'http://foo.io/aaa'}  | ${'bbb/'}               | ${'http://foo.io/aaa/bbb/'}
+    ${'http://foo.io/aaa/'} | ${'/bbb/'}              | ${'http://foo.io/aaa/bbb/'}
+    ${'http://foo.io/aaa/'} | ${'bbb/'}               | ${'http://foo.io/aaa/bbb/'}
+    ${'http://foo.io'}      | ${'http://bar.io/bbb'}  | ${'http://bar.io/bbb'}
+    ${'http://foo.io/'}     | ${'http://bar.io/bbb'}  | ${'http://bar.io/bbb'}
+    ${'http://foo.io/aaa'}  | ${'http://bar.io/bbb'}  | ${'http://bar.io/bbb'}
+    ${'http://foo.io/aaa/'} | ${'http://bar.io/bbb'}  | ${'http://bar.io/bbb'}
+    ${'http://foo.io'}      | ${'http://bar.io/bbb/'} | ${'http://bar.io/bbb/'}
+    ${'http://foo.io/'}     | ${'http://bar.io/bbb/'} | ${'http://bar.io/bbb/'}
+    ${'http://foo.io/aaa'}  | ${'http://bar.io/bbb/'} | ${'http://bar.io/bbb/'}
+    ${'http://foo.io/aaa/'} | ${'http://bar.io/bbb/'} | ${'http://bar.io/bbb/'}
+    ${'http://foo.io'}      | ${'aaa?bbb=z'}          | ${'http://foo.io/aaa?bbb=z'}
+    ${'http://foo.io'}      | ${'/aaa?bbb=z'}         | ${'http://foo.io/aaa?bbb=z'}
+    ${'http://foo.io/'}     | ${'aaa?bbb=z'}          | ${'http://foo.io/aaa?bbb=z'}
+    ${'http://foo.io/'}     | ${'/aaa?bbb=z'}         | ${'http://foo.io/aaa?bbb=z'}
+    ${'http://foo.io'}      | ${'aaa/?bbb=z'}         | ${'http://foo.io/aaa?bbb=z'}
+  `('$baseUrl + $x => $result', ({ baseUrl, x, result }) => {
     expect(resolveBaseUrl(baseUrl, x)).toBe(result);
   });
 
+  test.each`
+    baseUrl                 | x                       | result
+    ${'http://foo.io'}      | ${''}                   | ${'http://foo.io'}
+    ${'http://foo.io/'}     | ${''}                   | ${'http://foo.io'}
+    ${'http://foo.io'}      | ${'/'}                  | ${'http://foo.io/'}
+    ${'http://foo.io/'}     | ${'/'}                  | ${'http://foo.io/'}
+    ${'http://foo.io'}      | ${'/aaa'}               | ${'http://foo.io/aaa'}
+    ${'http://foo.io'}      | ${'aaa'}                | ${'http://foo.io/aaa'}
+    ${'http://foo.io/'}     | ${'/aaa'}               | ${'http://foo.io/aaa'}
+    ${'http://foo.io/'}     | ${'aaa'}                | ${'http://foo.io/aaa'}
+    ${'http://foo.io'}      | ${'/aaa/'}              | ${'http://foo.io/aaa/'}
+    ${'http://foo.io'}      | ${'aaa/'}               | ${'http://foo.io/aaa/'}
+    ${'http://foo.io/'}     | ${'/aaa/'}              | ${'http://foo.io/aaa/'}
+    ${'http://foo.io/'}     | ${'aaa/'}               | ${'http://foo.io/aaa/'}
+    ${'http://foo.io/aaa'}  | ${'/bbb'}               | ${'http://foo.io/bbb'}
+    ${'http://foo.io/aaa'}  | ${'bbb'}                | ${'http://foo.io/bbb'}
+    ${'http://foo.io/aaa/'} | ${'/bbb'}               | ${'http://foo.io/bbb'}
+    ${'http://foo.io/aaa/'} | ${'bbb'}                | ${'http://foo.io/bbb'}
```

Comment:
🤔 i would expect this to be `http://foo.io/aaa/bbb`

Reply:
It's `replaceUrlPath`, not `resolveBaseUrl`